src/java.naming/share/classes/com/sun/jndi/ldap/Connection.java

```diff
@@ -650,6 +650,21 @@ void cleanup(Control[] reqCtls, boolean notifyParent) {
                     } catch (IOException ie) {
                         if (debug)
                             System.err.println("Connection: problem closing socket: " + ie);
+                        //bug 8313657, socket not closed util GC running
+                        try {
+                            sock.close();
+                            unpauseReader();
+                            if (debug) {
+                                if (sock.isClosed()) {
```

Comment:
How is it possible for sock.close() to succeed but isClosed to return false?

Reply:
It is an unreachable code and got removed.